.github/workflows/sql-test-and-build-workflow.yml

```diff
@@ -57,11 +66,11 @@ jobs:
     - name: Upload Artifacts
       uses: actions/upload-artifact@v2
       with:
-        name: opensearch-sql
+        name: opensearch-sql-${{ matrix.os }}
         path: opensearch-sql-builds
 
     - name: Upload test reports
-      if: always()
+      if: ${{ matrix.os == 'ubuntu-latest' }}
```

Comment:
`if: always()` was there to have this step triggered even if previous failed. We have to see test reports when test failed.
Does this works: `if: ${{ always() && matrix.os == 'ubuntu-latest' }}`? You can damage any test to verify this.

Reply:
That worked, changed the PR